src/pages/variantHistory/ColumnHeaders.test.tsx

```diff
@@ -31,123 +35,150 @@ const initialState: HistoryTableReducerState = {
-            },
-          }),
-      }
+  it("should not show more column headers then the columnLimit", async () => {
+    const { Component } = RenderFakeToastContext(
+      <ColumnHeaders projectId="evergreen" variantName="some_variant" />
     );
-
-    await waitFor(() => {
-      expect(queryByText(longTaskName)).toBeNull();
+    const { queryAllByDataCy } = render(() => <Component />, {
+      wrapper: ({ children }) =>
+        wrapper({
+          children,
+          state: {
+            visibleColumns: ["task1", "task2", "task3", "task4", "task5"],
+            columnLimit: 3,
+          },
+          mocks: [mock(["task1", "task2", "task3", "task4", "task5"])],
+        }),
     });
     await waitFor(() => {
-      expect(queryByText("task2")).toBeVisible();
+      expect(queryAllByDataCy("loading-header-cell")).toHaveLength(0);
     });
+    expect(queryAllByDataCy("header-cell")).toHaveLength(3);
   });
+  it("should truncate the task name only if it is too long", async () => {
+    const { Component } = RenderFakeToastContext(
+      <ColumnHeaders projectId="evergreen" variantName="some_variant" />
+    );
+    const { queryByText, queryAllByDataCy } = render(() => <Component />, {
+      wrapper: ({ children }) =>
+        wrapper({
+          children,
+          state: {
+            visibleColumns: [longTaskName, "task2"],
+          },
+          mocks: [mock([longTaskName, "task2"])],
+        }),
+    });
 
+    await waitFor(() => {
+      expect(queryAllByDataCy("loading-header-cell")).toHaveLength(0);
+    });
+    expect(queryByText(longTaskName)).toBeNull();
+    expect(queryByText("task2")).toBeVisible();
+  });
   it("should show a tooltip with the full name when hovering over a truncated task name", async () => {
-    const { queryByText } = render(
-      () => (
-        <ColumnHeaders
-          loading={false}
-          projectId="evergreen"
-          columns={[longTaskName]}
-        />
-      ),
-      {
-        wrapper: ({ children }) =>
-          wrapper({
-            children,
-            state: {
-              visibleColumns: [longTaskName],
-            },
-          }),
-      }
+    const { Component } = RenderFakeToastContext(
+      <ColumnHeaders projectId="evergreen" variantName="some_variant" />
     );
+    const { queryByText, queryAllByDataCy } = render(() => <Component />, {
+      wrapper: ({ children }) =>
+        wrapper({
+          children,
+          state: {
+            visibleColumns: [longTaskName],
+          },
+          mocks: [mock([longTaskName])],
+        }),
+    });
+    await waitFor(() => {
+      expect(queryAllByDataCy("loading-header-cell")).toHaveLength(0);
+    });
     fireEvent.mouseEnter(queryByText(trimmedTaskName));
     await waitFor(() => {
       expect(queryByText(longTaskName)).toBeVisible();
     });
   });
 });
+
+const mock = (
+  buildVariants: GetTaskNamesForBuildVariantQuery["taskNamesForBuildVariant"]
```

Comment:
seems like this param should be named `taskNames`, not `buildVariants`? (for clarity)